### ISSUE TITLE
[FEATURE] Expose @parent variable in aggregate functions

### ISSUE THREAD

**Issue by QGIS-feature-tracker** [qgis/QGIS-Documentation#1478]
Original commit: https://github.com/qgis/QGIS/commit/f80a33be659200c1497a55c455a5521dc1e8c529 by m-kuhn

This makes it possible to access attributes and geometry from the parent
feature when in the filter of the "aggregate" expression function.

With this in place aggregates can be calculated per feature.

E.g. max "measurement" for each point_station per polygon_research_area.

Or a default attribute value when digitizing features:

```
aggregate(layer:='countries', aggregate:='max', expression:="code",
                  filter:=intersects( $geometry, geometry(@parent) ) )
```

**Comment by DelazJ** [comment]
@m-kuhn Is there a reason to not have @parent available under Variables section in the Expression builder?

**Comment by m-kuhn** [comment]
Hmm... I'd say 99% of the time it will not be meaningful, since it's only added in the context of an aggregate function. If we are going to add this it should possibly only be available when the cursor is inside an `aggregate` function (and combined with autocompletion?).

**Comment by DelazJ** [comment]
I suspected that reason. But I think it would gain in visibility (and it would deserve it) - its scope just needs to be clear in the help text.
I also wonder if it's possible to have it used in situations other than aggregate's, to write simpler expression that checks spatial or attribute relationship on a feature basis. Things like the reffunctions plugin does.
Autocompletion would be a great improvement, yeah! 
You see, there are things to do on expression side (not all are performance matters) at Essen and, btw  I just reported an issue on @parent http://hub.qgis.org/issues/16481 ;)
